### PR TITLE
Pivotal Slide Refactor [#130874755]

### PR DESCRIPTION
**Pull request by nc-choudharya**
Moved pivotal slide configurations to separate config file and decompose pivotal slide component into natural React components.

Also changed directory structure for easier navigation.

### CODE REVIEW

File: src/scripts/slides/pivotal_slide/components/pivotal_header.jsx [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+const PivotalHeader = React.createClass({
+  propTypes: {
+    headerTitle: React.PropTypes.string.isRequired
+  },
+
+  render() {
+    return (
+      <div className="static-header bb-row-table bb-r-container-content-block-align-center milestones-header">
+        <div
```
Comment (vsarang) [inline src/scripts/slides/pivotal_slide/components/pivotal_header.jsx]: NP: shorten to one line `<div className="header-container">`

File: src/scripts/slides/pivotal_slide/components/pivotal_milestone_bar.jsx [hunk 2]
```diff
@@ -0,0 +1,24 @@
+const React = require('react');
+
+const PivotalMilestoneBar = React.createClass({
+  propTypes: {
+    percentage: React.PropTypes.number.isRequired
+  },
+
+  render(){
+    const percentage = this.props.percentage;
```
Comment (vsarang) [inline src/scripts/slides/pivotal_slide/components/pivotal_milestone_bar.jsx]: NP: this variable is unnecessary

File: src/scripts/slides/pivotal_slide/components/pivotal_milestone_info.jsx [hunk 3]
```diff
@@ -0,0 +1,39 @@
+const React = require('react');
+
+const PivotalMilestoneInfo = React.createClass({
+  propTypes: {
```
Comment (vsarang) [inline src/scripts/slides/pivotal_slide/components/pivotal_milestone_info.jsx]: NP: let's alphabetize these props

File: src/scripts/slides/pivotal_slide/components/pivotal_milestone_info.jsx [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    milestone: React.PropTypes.string.isRequired,
+    completionDate: React.PropTypes.string.isRequired,
+    totalPoints: React.PropTypes.number.isRequired,
+    totalPointsCompleted: React.PropTypes.number.isRequired,
+    velocity: React.PropTypes.number.isRequired
+  },
+
+  render(){
+    const props = this.props;
```
Comment (vsarang) [inline src/scripts/slides/pivotal_slide/components/pivotal_milestone_info.jsx]: NP: none of these variables are needed; let's get rid of the clutter

File: src/scripts/slides/pivotal_slide/components/pivotal_milestones.jsx [hunk 5]
```diff
@@ -0,0 +1,122 @@
+const React = require('react');
```
Comment (vsarang) [inline src/scripts/slides/pivotal_slide/components/pivotal_milestones.jsx]: NP: `require`s should be alphabetized and grouped in order of externals then internals

```
const _ = require('lodash');
const axios = require('axios');
const moment = require('moment');
const Promise = require('es6-promise').Promise;
const React = require('react');

const API_TOKEN = require(...);

const PivotalMilestoneBar = require('./pivotal_milestone_bar.jsx');
const PivotalMilestoneInfo = require('./pivotal_milestone_info.jsx');
import config from './../pivotal_slide_config.js';
```

Also, could you look into the reason we aren't using `import` for all of these?

File: src/scripts/slides/pivotal_slide/components/pivotal_milestones.jsx [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+const PivotalMilestoneBar = require('./pivotal_milestone_bar.jsx');
+
+const PivotalMilestone = React.createClass({
+  propTypes: {
+    milestone: React.PropTypes.string.isRequired,
+    milestoneData: React.PropTypes.object.isRequired
+  },
+
+  render() {
+    const milestoneData = this.props.milestoneData;
```
Comment (vsarang) [inline src/scripts/slides/pivotal_slide/components/pivotal_milestones.jsx]: NP: again, these variables are unnecessary. Other than the most extreme cases, we shouldn't be using aliases to save us from typing.

File: src/scripts/slides/pivotal_slide/components/pivotal_milestones.jsx [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+          percentage={ percentage }
+        />
+      </div>
+    );
+  }
+});
+
+const PivotalMilestones = React.createClass({
+  fetchStoriesForMilestone(milestone, state) {
+    const url = config.BASE_URL + config.PROJECTS_URL + config.SEARCH +
```
Comment (vsarang) [inline src/scripts/slides/pivotal_slide/components/pivotal_milestones.jsx]: NP: tabbing is off, and you should be able to fit more on the first line:
```
const url = config.BASE_URL + config.PROJECTS_URL + config.SEARCH + milestone + state +
  config.NEGATIONS + ' includedone:true';
```

File: src/scripts/slides/pivotal_slide/components/pivotal_milestones.jsx [hunk 8]
```diff
@@ -0,0 +1,122 @@
...
+                milestone + state + config.NEGATIONS + ' includedone:true';
+    const self = this;
```
Comment (vsarang) [inline src/scripts/slides/pivotal_slide/components/pivotal_milestones.jsx]: This shouldn't be necessary if you're using an arrow function, right?

File: src/scripts/slides/pivotal_slide/components/pivotal_milestones.jsx [hunk 9]
```diff
@@ -0,0 +1,122 @@
...
+          [state]: response.data.stories
+        }
+      });
+
+      self.getVelocity(milestone, state);
+    });
+  },
+
+  getVelocity(milestone, state) {
+    const url = config.BASE_URL + config.PROJECTS_URL + config.SEARCH +
```
Comment (vsarang) [inline src/scripts/slides/pivotal_slide/components/pivotal_milestones.jsx]: NP: tabbing, same as above

File: src/scripts/slides/pivotal_slide/components/pivotal_milestones.jsx [hunk 10]
```diff
@@ -0,0 +1,122 @@
...
+                milestone + config.NORMALIZE_VELOCITY + config.NEGATIONS +
+                ' includedone:true';
+    const self = this;
```
Comment (vsarang) [inline src/scripts/slides/pivotal_slide/components/pivotal_milestones.jsx]: Same as above, this shouldn't be necessary.

File: src/scripts/slides/pivotal_slide/components/pivotal_milestones.jsx (hunk 6 above)
Comment (nc-choudharya) [inline src/scripts/slides/pivotal_slide/components/pivotal_milestones.jsx]: Doesn't it help in readability if something like 
```
this.that.which.foo = 5;
this.that.which.bar = 6;
```
is written as
```
let it = this.that.which;
it.foo = 5;
it.bar = 6;
```

Comment (vsarang) [inline src/scripts/slides/pivotal_slide/components/pivotal_milestones.jsx]: Well usually it is more like
```
let it = this.that.which;

// block of code

it.foo = 5;
it.bar = 6;
```
And then you have to scroll up and find where `it` was defined to check what it is. This is made worse when different devs use different names for their variables (so we have the same object being mapped to different variable names in different files, hurting readability) and when this happens multiple times in the same file (and you now have to keep track of multiple variables that have been remapped to different names).

**Comment by vsarang** [comment]
@nc-choudharya Is this ready for a second pass? Go ahead and dismiss the review or reassign me if so.

**Comment by nc-choudharya** [comment]
@vsarang About the imports not being in es6 style, I think it's probably because gulp isn't converting it into regular es5, which would mean that the code might break on old browsers (not sure what browser is used on the screen where the dashboard is displayed)
Let me know when you've taken a second look at it. I'll squash the commits then.

**Pull request closed without merging**